src/uri.rs

```diff
@@ -2,6 +2,9 @@ use std::{hash::Hash, ops::Deref, str::FromStr};
 
 use serde::{de::Error, Deserialize, Serialize};
 
+mod canonical;
+pub use canonical::*;
```

Comment:
Was this intentional?
```shell
warning: unused import: `canonical::*`
 --> src/uri.rs:6:9
```

```suggestion
```

Reply:
Oh, no. I completely forgot about this PR. That commit is just something I use by myself and wasn't meant to be pushed upstream.